detection_rules/config.py

```diff
@@ -41,6 +42,44 @@ def __post_init__(self):
         assert not (self.bypass and self.test_only), 'Cannot use both test_only and bypass'
 
 
+@dataclass
+class ConfigFile:
+    """Base object for configuration files."""
+
+    @dataclass
+    class FilePaths:
+        packages_file: str
+        stack_schema_map_file: str
+        deprecated_rules_file: Optional[str] = None
+        version_lock_file: Optional[str] = None
+
+    @dataclass
+    class TestConfigPath:
+        config: str
+
+    files: FilePaths
+    rule_dir: List[str]
+    testing: Optional[TestConfigPath] = None
+
+    @classmethod
+    def from_dict(cls, obj: dict):
```

Comment:
```suggestion
    def from_dict(cls, obj: dict) -> ConfigFile:
```

Reply:
its not instantiated at this point, but we can still add some hinting.